desktop/cmp/button/RefreshButton.js

```diff
@@ -9,41 +9,66 @@ import {Component} from 'react';
 import {PropTypes as PT} from 'prop-types';
 import {elemFactory, HoistComponent} from '@xh/hoist/core';
 import {Icon} from '@xh/hoist/icon';
-import {button} from '@xh/hoist/kit/blueprint';
+import {button} from './Button';
+import {warnIf} from '@xh/hoist/utils/JsUtils';
 
 /**
  * Convenience Button preconfigured for use as a trigger for a refresh operation.
  * Accepts props documented below as well as any others supported by Blueprint's Button.
  *
- * Must be provided either an onClick handler *or* a model. If a model is provided, this button
- * will call loadAsync() on the model class.
+ * Must be provided either an onClick handler *or* a model. If a model is provided and an onClick
+ * handler is not provided, this button will call loadAsync() on the model class.
  */
 @HoistComponent()
 export class RefreshButton extends Component {
 
     static propTypes = {
+        /** Icon to display for the button. Defaults to Icon.sync(). */
         icon: PT.element,
+        /** Tooltip text to display when the mouse is over the button. Defaults to 'Refresh'. */
         title: PT.string,
+        /** Function to call when the button is clicked. */
         onClick: PT.func,
+        /**
+         * Model to call loadAsync() on when the button is clicked. Only used if onClick was not
+         * provided.
+         */
         model: PT.object
     };
 
     render() {
-        const {icon, title, onClick, model, ...rest} = this.props;
+        warnIf(
+            this.props.model && this.props.onClick,
+            'Both model and onClick were provided to RefreshButton properties. The model property will be ignored.'
+        );
+
+        warnIf(
+            !this.props.model && !this.props.onClick,
+            'Neither onClick nor model were provided to RefreshButton. Clicking the RefreshButton will have no effect.'
+        );
```

Comment:
Deciding how/when we want to add these kinds of warnings will be more of an art than a science. It's a balance between helping to document the component and avoid common pitfalls, without adding too much defensive code. 

I'm glad you used the pattern here, but think a slightly better balance would be to collapse these into one check that could output something like "RefreshButton must be provided either a model or an onClick handler to call (but not both)."